### PR TITLE
Cat Harvest for the Scene System

### PR DESCRIPTION
**Pull request by ghost**
This is a mostly faithful port of Cat Harvest into the Scene System, I've made changes to the way particles spawn to be better visible, and gave the planted cats an occasional particle burst to better indicate where they are every 25 seconds. 
All endings should be possible.
If any errors come up leave an issue and I'll maintain it 

oh and make sure you change the org and indent to whatever

### CODE REVIEW

**Comment by ghost** [comment]
I've added Stats (and Achievements) as mentioned as TODO by this comment
`Log.Info( "TODO: scoreboard" );`
Here's the Stat Indents
```
total_cats_uprooted   | Increments by 1 for each cat you uproot
cats_harvested        | Increments by 1 for each cat you harvest
cats_rescued          | Increments by 1 for each cat you rescue
```
Here's the Achievement Indents
```
balanced_ending 
genocide_ending
neutral_ending
peaceful_ending
secret_ending
```

File: .gitattributes
Comment (rndtrash) [inline .gitattributes:1]: why?

File: Assets/scenes/minimal.scene
Comment (rndtrash) [inline Assets/scenes/minimal.scene:1]: `minimal.scene` please rename it to something else, like `game.scene`

File: Code/Endings.cs [hunk 1]
```diff
@@ -210,68 +212,65 @@ public static async void SecretEnding( HarvestPlayer ply )
...
 
 		current.Jumpscare = true;
-		Sound.FromScreen( "angry0" ).SetVolume( 255f );
-		Sound.FromScreen( "angry0" ).SetVolume( 1f );
-		Sound.FromScreen( "angry0" ).SetVolume( 10f );
-		Sound.FromScreen( "angry0" ).SetVolume( 50f ); //Just to be sure, not sure why it doesn't play sometimes
+		Sound.Play( "angry0" ).Volume = 255f;
+		Sound.Play( "angry0" ).Volume = 1f;
+		Sound.Play( "angry0" ).Volume = 10f;
+		Sound.Play( "angry0" ).Volume = 50f; //Just to be sure, not sure why it doesn't play sometimes
```
Comment (rndtrash) [inline Code/Endings.cs:229]: are you sure this is necessary?

File: Code/Endings.cs [hunk 2]
```diff
@@ -210,68 +212,65 @@ public static async void SecretEnding( HarvestPlayer ply )
...
-
+		Sandbox.Services.Achievements.Unlock( "secret_ending" );
 		await current.Task.Delay( 2000 );
 
+		//real funny
+		if ( Game.IsEditor )
+		{
+			Game.Close();
+			return;
+		}
```
Comment (rndtrash) [inline Code/Endings.cs:239]: well, now I think it's safe to remove the while ( true ) loop

File: Code/MyComponent.cs
Comment (rndtrash) [inline Code/MyComponent.cs:1]: please remove the placeholders

File: Code/HarvestGame.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		SecretCat.IsSecret = true;
+		Music = Sound.Play( "sounds/relax.sound" );
+	}
+
+	protected override void OnFixedUpdate()
+	{
+		base.OnFixedUpdate();
+
+		var allObjects = Scene.GetAllObjects(true)
+			.Where( x => x.Tags.Has( "cat") && x.GetComponent<WalkingCat>().IsValid() );
```
Comment (rndtrash) [inline Code/HarvestGame.cs:38]: there should be a better way, wasn't there a method to find every object in a scene by a component?

File: Code/HarvestGame.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		
+		AllCats = allObjects.ToList();
+		
+		TheSnappening();
```
Comment (rndtrash) [inline Code/HarvestGame.cs:42]: ...the what???

File: Code/Player/HarvestPlayer.cs [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+				{
+					cat.Destroy();
+
+					GetComponentInChildren<SkinnedModelRenderer>().Set( "wiwi", true );
+					HarvestGame.EndGame( this, 0, true );
+				}
+			}
+			else
+			{
+				if ( cat.GetComponent<WalkingCat>().IsValid() ) return;
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:77]: checking the cat validness twice here

File: Code/Player/HarvestPlayer.cs [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+				Popup = PopupType.Uproot;
+
+				if ( Input.Pressed( "use" ) )
+				{
+					Sound.Play( $"meow{Game.Random.Int( 10 )}", cat.WorldPosition );
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:82]: i think you can combine all the meows into one sound file and then use s&box's built in random sound player mode

File: Code/Player/HarvestPlayer.cs [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+					var particle = LegacyParticle.Create( "particles/uproot.vpcf", cat.WorldPosition );
+						particle.GameObject.DestroyAsync(3);
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:84]: magic constant here

File: Code/Player/HarvestPlayer.cs [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+		}
+		else
+		{
+			Popup = PopupType.None;
+		}
+
+		if ( _playerController.Velocity.Length > 0f && lastStep >= 70 / _playerController.Velocity.Length && _playerController.IsOnGround )
+		{
+			var step = $"step{Game.Random.Int( 5 )}";
+			Sound.Play( step, WorldPosition );
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:104]: same as the meows

File: Code/Player/HarvestPlayer.cs [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+	public static void Harvest()
+	{
+		var ply = Game.ActiveScene.Scene.GetComponentInChildren<HarvestPlayer>();
+		if ( !ply.IsValid() ) return;
+		ply.CatsHarvested++;
+		ply.HasCat = false;
+		ply.GetComponentInChildren<SkinnedModelRenderer>().Set("finished", true);
+		Sandbox.Services.Stats.Increment("cats_harvested",1);
+		
+		Sound.Play( $"sad{Game.Random.Int( 1 )}", ply.WorldPosition );
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:118]: same here

File: Code/Player/HarvestPlayer.cs [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+
+		if ( _playerController.Velocity.Length > 0f && lastStep >= 70 / _playerController.Velocity.Length && _playerController.IsOnGround )
+		{
+			var step = $"step{Game.Random.Int( 5 )}";
+			Sound.Play( step, WorldPosition );
+			lastStep = 0f;
+		}
+	}
+
+	public static void Harvest()
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:109]: should it really be static?

File: Code/Player/HarvestPlayer.cs [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+		ply.HasCat = false;
+		ply.GetComponentInChildren<SkinnedModelRenderer>().Set("finished", true);
+		Sandbox.Services.Stats.Increment("cats_harvested",1);
+		
+		Sound.Play( $"sad{Game.Random.Int( 1 )}", ply.WorldPosition );
+		var particle = LegacyParticle.Create( "particles/dollars.vpcf", ply.WorldPosition + Game.ActiveScene.Scene.Camera.WorldRotation.Forward * 32f + ply.WorldRotation.Up * 48f);
+		particle.GameObject.DestroyAsync(3);
+		if ( ply.CatsUprooted == 96 )
+		{
+			HarvestGame.EndGame( ply, ply.CatsHarvested );
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:123]: might as well pass only the player reference

File: Code/Player/HarvestPlayer.cs [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+		}
+	}
+	
+	public static void Rescue()
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:127]: same here

File: Code/Player/HarvestPlayer.cs [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+		{
+			HarvestGame.EndGame( ply, ply.CatsHarvested );
+		}
+	}
+
+	
+	public void OnKilled()
+	{
+		//Don't die! wtf
+	}
```
Comment (rndtrash) [inline Code/Player/HarvestPlayer.cs:148]: :) that's a relic of the entity days, leave it if you want

File: Editor/MyEditorMenu.cs
Comment (rndtrash) [inline Editor/MyEditorMenu.cs:1]: we don't need no editor tools

File: ProjectSettings/Input.config [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+      "GroupName": "Inventory"
+    },
+    {
+      "Name": "Slot8",
+      "KeyboardCode": "8",
+      "GamepadCode": "None",
+      "GroupName": "Inventory"
+    },
+    {
+      "Name": "Slot9",
```
Comment (rndtrash) [inline ProjectSettings/Input.config:124]: can you please delete the unused inputs?

File: cat_harvest.sbproj [hunk 15]
```diff
@@ -0,0 +1,27 @@
+{
+  "Title": "Cat Harvest",
+  "Type": "game",
+  "Org": "local",
```
Comment (rndtrash) [inline cat_harvest.sbproj:4]: local -> fish

File: cat_harvest.sln.DotSettings.user
Comment (rndtrash) [inline cat_harvest.sln.DotSettings.user:1]: shouldn't this file be ignored?

File: Code/UI/HarvestHUD.cs
Comment (rndtrash) [inline Code/UI/HarvestHUD.cs:1]: HUD: can you please rewrite it in Razor? Also I think it's safe to pass the player reference to the panel component, thus there won't be nonneed to look for the player component every frame

File: Code/Endings.cs [hunk 16]
```diff
@@ -1,12 +1,13 @@
...
 
 namespace CatHarvest;
 
 public partial class HarvestGame
 {
-	[Net] public bool EndState { get; set; } = false;
-	[Net] public int Ending { get; set; } = 0;
-	[Net] public bool Jumpscare { get; set; } = false;
+	public bool EndState { get; set; } = false;
+	public int Ending { get; set; } = 0;
```
Comment (rndtrash) [inline Code/Endings.cs:9]: can you make it an enum?

File: Code/Catcheck.cs
Comment (rndtrash) [inline Code/Catcheck.cs:1]: is it for debugging? either way I think that HarvestGame has this case covered, should be safe to get rid of

File: Code/Cameras.cs [hunk 17]
```diff
@@ -0,0 +1,42 @@
+﻿using Sandbox.Entities;
+
+namespace CatHarvest;
+
+public class EndingCam : Component
+{
+	[Property] public CameraComponent Camera;
+	public TimeSince tsCreated;
+	public bool IsPeacefulEnding = false;
+	public bool IsBalancedEnding = false;
```
Comment (rndtrash) [inline Code/Cameras.cs:10]: can you check the ending straight from the HarvestGame component?

File: Assets/test.prefab
Comment (rndtrash) [inline Assets/test.prefab:1]: test? what does it do?

**Review by rndtrash — CHANGES_REQUESTED** [review]
I'm on the phone right now, that's as much as I can review. Either way, good job! Cat Harvest is a certified s&box classic that should be made playable once again :)

**Pull request merged**